_adr/90/index.adoc

```diff
@@ -0,0 +1,124 @@
+
+## Proposal
+
+The proposed solution is going to use https://Letsencrypt.org[Letsencrypt] to automatically manage certificates.
+The solution consists of: 
+
+1. Generating one wildcard certificate per Kafka instance .
+For each new `standard` Kafka instance, there will be a wildcard certificate for the `*.<kafka-id>.kafka.bf2.dev` subdomain where `<kafka-id>` is unique for each Kafka.
+For each new `developer` Kafka instance, a shared wildcard certificate `*.trial.kafka.bf2.dev` used for all developer instances will be generated and automatically managed. 
+
+2. The fleet manager in its reconcilers will use https://github.com/caddyserver/certmagic[certmagic library] to generate certificate for each new Kafka and store them in a secure Storage. 
+During Kafka reconciliation between the Fleetshard and the Fleet manager, certificates will be fetched from the secure storage by the fleet manager and they'll be passed down to the data plane via the existing certificate passing mechanism i.e via the ManagedKafka CR.
+The fleet manager will implement this logic in a way that it can be easily extracted & packaged in form of a library and shared with others managed services builders.
+The kas-fleet-manager certificate reconciliation will have to handle the two different cases below: 
+a. When a `standard` kafka is assigned to a given data plane, all routes created will be under the `*.<kafka-id>.kafka.bf2.dev` domain. 
+For example, if the kafka id is `12345` then Kafka boostrap host will have the form of `foobar-xsqhyyza.12345.kafka.bf2.dev`.
+b. When a `developer` kafka is assigned to a given data plane, all routes created will be under the `*.trial.kafka.bf2.dev` domain. 
+For example, if the kafka id is `12345` then Kafka boostrap host will have the form of `foobar-sqfsazk.trial.kafka.bf2.dev`.
+
+3. No migration to be performed for Kafkas.
+The proposed solution will be applied to only new Kafkas.  
+Existing Kafkas will continue to share the wildcard certificate on the `*.kafka.bf2.dev` domain.  
+The shared certificate for the already existing Kafkas will now be automatically renewed by the fleet manager.
+
+4. Certificate renewals will also be handled automatically by the fleet manager using the https://github.com/caddyserver/certmagic[certmagic library]. 
+Renewals will be blast radius aware to renew certificates in small runs on per Kafka basis.
+Renewals of each certificate will happen 30 days before the certificate expires. 
+The intention is to reduce the risk of compromising all our certificates in case there is an outage or an issue with the renewal system in Letsencrypt. 
+It also removes the chances of bombarding Letsencrypt with bulk requests or rollout of all the Kafkas in one big batch after renewals. 
+
+5. Revocation will be handled per Kafka instance.
```

Comment:
revocation is an interesting area. As I understand it, it relies on the client actually checking if it has been revoked. From some brief reading, this seems to often not be the case. https://github.com/cert-manager/cert-manager/issues/392 . Below seems to describe replacing a compromised cert rather than actively revoking it and adding it to the LE revoked cert list? https://letsencrypt.org/docs/revoking/

Reply:
The way that the flow work is that it'll actually call LE and revoke the certificate with `keyCompromised` reason.